task/buildah/0.3/README.md

```diff
@@ -0,0 +1,63 @@
+# Buildah
+
+This Task builds source into a container image using Project Atomic's
+[Buildah](https://github.com/projectatomic/buildah) build tool. It uses
+Buildah's support for building from
+[`Dockerfile`](https://docs.docker.com/engine/reference/builder/)s, using its
+`buildah bud` command. This command executes the directives in the `Dockerfile`
+to assemble a container image, then pushes that image to a container registry.
+
+## Install the Task
+
+```
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/task/buildah/0.3/buildah.yaml
+```
+
+## Parameters
+
+* **IMAGE**: The name (reference) of the image to build.
+* **BUILDER_IMAGE:**: The name of the image containing the Buildah tool. See
+  note below.  (_default:_ quay.io/buildah/stable:v1.15.1)
+* **DOCKERFILE**: The path to the `Dockerfile` to execute (_default:_
+  `./Dockerfile`)
+* **CONTEXT**: Path to the directory to use as context (_default:_
+  `.`)
+* **TLSVERIFY**: Verify the TLS on the registry endpoint (for push/pull to a
+  non-TLS registry) (_default:_ `true`)
+* **FORMAT**: The format of the built container, oci or docker (_default:_
+ `oci`)
+* **BUILD_EXTRA_ARGS**: Extra parameters passed for the build command when
+  building images. (_default:_ `""`)
+* **PUSH_EXTRA_ARGS**: Extra parameters passed for the push command when
+  pushing images. (_default:_ `""`)
+* **STORAGE_DRIVER**: Set build storage driver to use (_default:_ `"vfs"`_)
```

Comment:
I wonder if we should have this parameter at all (and if we should document it or not)

Reply:
Isnt there some drawbacks using vfs ? Being much slower? (maybe i am confused tho which option is the really slow one here)

The user may want to adjust it depend of their build workload and not being bothered about running it as Root

Reply:
> Isnt there some drawbacks using vfs ? Being much slower? (maybe i am confused tho which option is the really slow one here)

Well, usually there is, but in a container in a cluster, the difference is very very very small (cc @gabemontero). It is small enough that we cane consider using `vfs` by default, and thus allowing us to not required `privileged`.

Reply:
I mean the drawback to don't give the choise the user with a  params.

Reply:
Ah :sweat_smile: well keeping it means it might or might not work (if the user provides `overlay` for `STORAGE_DRIVER` without being able to run it as `privileged` it will fail)

Reply:
there have also been enhancements in buildah wrt vfs since this example was first constructed

some recent experimentation on our part in OCP showed the difference now was not discernable.

IIRC I believe historically this example was using 1.11

but I see for example @vdemeester 's update moves to `quay.io/buildah/stable:v1.15.1`

also, IIRC, @nalind from the buildah team suggested 1.16 for even more improvements ... I would consider that change as well @vdemeester 

And moving forward, I think we should engage with him directly on these types of questions, which I've attempted to do here of course :-) ... please consider this the "formal introduction" @vdemeester @chmouel .... I'll mention this to him in the next meeting I'm in with him as well.

Reply:
1.16 should handle ADD and COPY better (faster) than it used to, and we've made correctness fixes as we've gone along.

Keep in mind that most of the gains we're seeing with vfs have happened because it tries to take advantage of the FICLONE and FICLONERANGE ioctls when copying the contents of one layer to create the basis for another layer, to which it then applies a layer diff to finish initializing the new layer's contents.  That ioctl() call will only succeed if the underlying filesystem supports it.  For OpenShift nodes, we can generally expect it to be XFS formatted with "-m reflink=1", which provides that ability, but I can't make guarantees about clusters in general.

Reply:
Ah .... very informative / useful @nalind  - thanks

so @vdemeester @chmouel @sbose78 @siamaksade .... vs. how we went with downstream openshift pipelines, perhaps we want to consider exposing the "performance vs. security" nuances around buildah in our upstream examples (as the performance impact is possibly more likely on non-openshift nodes)?

@sbose78 I know this was on your radar in particular recently ... thoughts?

Reply:
thanks @nalind and @gabemontero, yeah maybe a documentation update mentionning nalind comment would be great for our users,

Reply:
I'm good as long as the default is vfs. If an advanced user wishes to flip that to "overlay", I would expect the advanced user to read documentation where the nuances (incl need for higher privileges) would be explained. :)